src/lib/rest/RestService.cpp

```diff
@@ -52,9 +52,111 @@
 #define MAX_TENANT_NAME_LEN_STRING    "50"
 
 
+static std::string textParseAttributeValue(ConnectionInfo* ciP, ContextAttribute* caP)
+{
+  double d;
+
+  // 1. Starts and ends with citation marks?
+  if (ciP->payload[0] == '"')
+  {
+    char* end = &ciP->payload[strlen(ciP->payload) - 1];
+
+    if (*end == '"')
+    {
+      *end = 0;
+      caP->stringValue = &ciP->payload[1];
+      caP->valueType   = orion::ValueTypeString;
+    }
+    else
+    {
+      OrionError oe(SccBadRequest, "Missing citation-mark at end of string");
+
+      return oe.render(ciP, "");
+    }
+  }
+
+  // 2. True or false?
+  else if ((strlen(ciP->payload) == 4) && ((strcmp(ciP->payload, "true") == 0) || (strcmp(ciP->payload, "True") == 0) || (strcmp(ciP->payload, "TRUE") == 0)))
+  {
+    caP->boolValue   = true;
+    caP->valueType   = orion::ValueTypeBoolean;
+  }
+  else if ((strlen(ciP->payload) == 5) && ((strcmp(ciP->payload, "false") == 0) || (strcmp(ciP->payload, "False") == 0) || (strcmp(ciP->payload, "FALSE") == 0)))
+  {
+    caP->boolValue   = false;
+    caP->valueType   = orion::ValueTypeBoolean;
+  }
+
+  // 3. Null ?
+  else if ((strlen(ciP->payload) == 4) && ((strcmp(ciP->payload, "null") == 0) || (strcmp(ciP->payload, "Null") == 0) || (strcmp(ciP->payload, "NULL") == 0)))
+  {
+    caP->valueType   = orion::ValueTypeNull;
+  }
+
+  //
+  // 4. Valid Double?
+  // FIXME P4: this is much more complex than just (atof(string) != 0 || string == "0")
+  //           0.000 also is a valid float and it given 0 - 0.0000000000, and 0e0 also ...
+  //           And, even worse, 123K gives 123.0 back, we would need to analyze the string to try to
+  //           find garbage bytes after it if we want to detect this error.
+  //           However, all of this is not so extremely important and for now, (atof(string) != 0 || string== "0") is good enough
+  //
+  else if (((d = strtod(ciP->payload, NULL)) != 0) || ((ciP->payload[0] == '0') || (ciP->payload[1] == 0)))
+  {
+    caP->valueType   = orion::ValueTypeNumber;
+    caP->numberValue = d;
+  }
+
+  // 5. None of the above - it's an error
+  else
+  {
+    OrionError oe(SccBadRequest, "attribute value type not recognized");
+    return oe.render(ciP, "");
+  }
+
+  return "OK";
+}
+
+
+
+/* ****************************************************************************
+*
+* textRequestTreat - to be moved to ... own library?
+*
+* FIXME P10: Where do we put this function?
```

Comment:
I put P10 to make sure it is fixed in this PR, not because it was especially important per se.
We have three different libraries already for parsing stuff, xmlParse and two for jsonParse.
To follow our own style, we should make a new library for 'textParse'.

However, it is a little bit overkill now that we have one single function ...

There is a library containing common functions for the different parse libraries (the library is called **parse**), but that is also not the place for this test-parsing function ...

If we want to do it "well", we add a new library, called textParse, for example.
If not ... then the best place for this function is right where it resides right now (rest/RestService.cpp), IMO.